src/types/InvertPattern.ts

```diff
@@ -18,6 +18,8 @@ export type InvertPattern<p> = p extends typeof __.number
   ? string
   : p extends typeof __.boolean
   ? boolean
+  : p extends typeof __.unit
+  ? null | undefined
```

Comment:
I'm a bit afraid by the compile time perf implication of this change. I know it's just another case, but I think we should try to remove cases instead of adding new ones

Reply:
I was worried about this.